src/content/app/genome-browser/state/browser-entity/browserEntitySlice.ts

```diff
@@ -0,0 +1,219 @@
+    browserStorageService.updateChrLocation({ [activeGenomeId]: chrLocation });
+
+  if (activeEnsObjectId) {
+    browserStorageService.updateActiveEnsObjectIds({
+      [activeGenomeId]: activeEnsObjectId
+    });
+  }
+};
+
+export const updateBrowserActiveEnsObjectIdsAndSave = (
+  activeEnsObjectId: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const currentActiveEnsObjectIds = getBrowserActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds,
+      [activeGenomeId]: activeEnsObjectId
+    };
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+    dispatch(fetchEnsObject(activeEnsObjectId));
+
+    browserStorageService.updateActiveEnsObjectIds(updatedActiveEnsObjectIds);
+  };
+};
+
+export const deleteSpeciesInGenomeBrowser = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+
+    dispatch(deleteGenomeEntityData(genomeIdToRemove));
+    dispatch(deleteGenomeLocationData(genomeIdToRemove));
+    dispatch(deleteGenomeTrackPanelData(genomeIdToRemove));
+
+    const updatedActiveEnsObjectIds = pickBy(
+      getBrowserActiveEnsObjectIds(state),
+      (value, key) => key !== genomeIdToRemove
+    );
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.deleteGenome(genomeIdToRemove);
+    trackPanelStorageService.deleteGenome(genomeIdToRemove);
+  };
+};
+
+const browserEntitySlice = createSlice({
+  name: 'genome-browser-entity',
+  initialState: defaultBrowserEntityState as BrowserEntityState,
+  reducers: {
+    setActiveGenomeId(state, action: PayloadAction<string>) {
+      const genomeId = action.payload;
+      state.activeGenomeId = genomeId;
+    },
+    setEntityDataFromUrl(state, action: PayloadAction<ParsedUrlPayload>) {
+      const { activeGenomeId, activeEnsObjectId } = action.payload;
+
+      state.activeGenomeId = activeGenomeId;
+      if (activeEnsObjectId) {
+        state.activeEnsObjectIds[activeGenomeId] = activeEnsObjectId;
+      }
+    },
+    updateBrowserActiveEnsObjectIds(
+      state,
+      action: PayloadAction<{ [objectId: string]: string }>
+    ) {
+      state.activeEnsObjectIds = action.payload;
+    },
+    updateTrackStates(state, action: PayloadAction<BrowserTrackStates>) {
+      state.trackStates = merge({}, state.trackStates, action.payload);
+    },
+    deleteGenomeEntityData(state, action: PayloadAction<string>) {
+      const genomeIdToRemove = action.payload;
+      const activeGenomeId = state.activeGenomeId;
+
+      if (activeGenomeId === genomeIdToRemove) {
+        state.activeGenomeId = null;
+        state.activeEnsObjectIds = pickBy(
```

Comment:
just `delete state.activeEnsObjectIds[genomeIdToRemove]` ?

Reply:
That's done